template/script.sh.erb

```diff
@@ -47,24 +42,25 @@ EOL
 )
 chmod 700 "${RSESSION_WRAPPER_FILE}"
 
-# Define launcher for `rserver` if not using Singularity
-if [[ -z "${RSTUDIO_SINGULARITY_IMAGE}" ]]; then
-  PROOT_BIN="${PROOT_BIN:-"/usr/local/proot/bin/proot-x86_64"}"
-  RSERVER_LAUNCHER="${PROOT_BIN} -b $(mktemp -d):/tmp "
-fi
-
 # Set working directory to home directory
 cd "${HOME}"
 
 # Output debug info
 module list
 
+# Create tmpdir rooted in PBSTMP
+# This will be bound into the guest and mounted at /tmp
+CONTAINER_TMP=$(mktemp -d)
```

Comment:
I don't think this has anything to do with $PBSTMP.  If you wanted to use environment variables you could use `$TMPDIR` which is setup by Torque to be a job-specific directory in /tmp.  You would have to ensure it gets set within the container via something like this:

```
export SINGULARITYENV_TMPDIR=$TMPDIR
```

Reply:
That could be that `PBSTMP` is something else. The way what is written works is the guest sees `/tmp`, which is mounted on the host at something like `/tmp/pbs.BLAH/tmp.MORE_BLAH`, which is harmlessly redundant.

That said, I suppose that if there is an environment variable specifically for setting the temp dir in the guest it would be more semantically correct to use that instead of another bind.